### DISCUSSION TITLE
Set PR merge config by @MunifTanjim

### DISCUSSION THREAD

**Discussion by gr2m** [octoherd/octoherd#18] [Show and tell]
Hey @MunifTanjim I just found out about 
https://github.com/MunifTanjim/octoherd-script-set-pr-merge-config

Great to see people creating @octoherd scripts and sharing them with others, thank you!

How did the setup go? Did you use `npm init octoherd-script` to set it up? How was the experience? Anything we could improve? Any other thoughts?

**Comment by MunifTanjim** [comment]
> How did the setup go? Did you use npm init octoherd-script to set it up?

Yes! 😃 

> How was the experience?

It was absolutely smooth. I probably did the whole packge within 5-10 minutes. Great work on `octoherd-script` 🎉 

> Anything we could improve? Any other thoughts?

I noticed a minor bug, but forgot to report it.

https://github.com/MunifTanjim/octoherd-script-set-pr-merge-config/blob/54ae9c37370a3dd8ac32925a0672e0b2f9b2743a/script.js#L9-L11

After the code generation, these 3 lines had `options.undefined`.

And one other thing... So, I needed to explicitly set a flag value to `false` for the cli, but it wasn't obvious how to do that. I had to dig into the source code to find out it uses `yargs` and then found this: https://github.com/yargs/yargs/blob/1a1e2d554dca3566bc174584394419be0120d207/docs/tricks.md#negating-boolean-arguments

Maybe it'll be helpful mentioning in the README that `@octoherd/cli` uses `yargs` along with the link to its documentation?

**Reply by gr2m** [reply]
Glad it worked so well, yay 🎉 thank you for sharing!

I'll create follow up issues for the problems you mentioned. Just so I understand correctly, instead of

```
 * @param {boolean} [options.mergeCommit]
 * @param {boolean} [options.rebaseMerge]
 * @param {boolean} [options.squashMerge]
```

the lines said

```
 * @param {boolean} [octokit.undefined]
 * @param {boolean} [octokit.undefined]
 * @param {boolean} [octokit.undefined]
```

or was it

```
 * @param {boolean} [options.undefined]
 * @param {boolean} [options.undefined]
 * @param {boolean} [options.undefined]
```

I think I know what the problem is

**Reply by MunifTanjim** [reply]
Ah, sorry, there was a typo. It was `options.undefined`
